api/models.py

```diff
@@ -158,6 +155,27 @@ class Revision(BaseModel):
     )
 
 
+class DefaultTimeout:
+    """Helper to create default values for timeout fields"""
+
+    def __init__(self, hours=0, minutes=0, delta=None):
+        """Set the timeout delta
+
+        If `delta` is provided, use that directly as a timedelta object.
+        Otherwise, construct one with the `hours` and `minutes` values.
+        """
+        self._delta = delta or timedelta(hours=hours, minutes=minutes)
+
+    @property
+    def delta(self):
+        """Get the timedelta set in this object"""
+        return self._delta
+
+    def get_timeout(self):
+        """Get a timeout timestamp with current time and delta"""
+        return datetime.utcnow() + self.delta
```

Comment:
Shouldn't we use `created` timestamp to calculate timeout?

Reply:
I'm not sure we can enforce `created` to be initialised first, and for similar reasons we can't pass it to the default_factory callable in the class definition.  In practice, it'll still be "now" between when the two fields are initialised.  If we ever get to the point where there's more than one minute difference between created and the time used to calculate the timeout then I think we'll have much more serious issues to address.

In case the client is providing the `created` time but no timeout while submitting a new node, then the timeout will most likely be more than 6h after creation time.  But I don't think that's an issue, and it's kind of expected.

Reply:
I've added a `start_time` argument so the caller could pass the `created` time or another start time if that's needed.  If not it will keep using `.utcnow()` by default.

Reply:
Actually I'm not sure we'll ever be able to use `start_time` with the original intention to use this class when defining models as we can't pass any values to it.  I was thinking we might be able to use it for `holdoff` when the node goes to `Available` state and the client hasn't specified a holdoff timeout, if we need to set a default one too.  If we don't need to set any default then I'll revert to the previous implementation with no `start_time`.

Reply:
I am not able to think of any use case when a caller would need to pass the `start_time`.
Even how will the `delta` be dynamic with this implementation?
We need to receive it from the request parameter to make it client specific.

Reply:
Right, so I'll revert to the previous implementation and always rely on `.utcnow()` as the start time.